source/Halibut.TestUtils.CompatBinary.Base/Utils/DataStreamExtensionMethods.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.IO;
+
+namespace Halibut.TestUtils.SampleProgram.Base.Utils
+{
+    public static class DataStreamExtensionMethods
+    {
+        public static void ConfigureWriterOnReceivedDataStream(this DataStream dataStream)
+        {
+            Action<Stream> writer = stream => { dataStream.Receiver().Read(dataStreamStream => dataStreamStream.CopyTo(stream)); };
+            dataStream.ReflectionSetFieldValue("writer", writer);
+        }
+    }
+}
```

Comment:
I'm going to need you to explain this to me slowy 😄

Reply:
Added a comment on what is going on.